test/nodetool/utils.py

```diff
@@ -13,17 +13,21 @@ def check_nodetool_fails_with(
         nodetool,
         nodetool_args: tuple,
         nodetool_kwargs: dict,
-        expected_errors: typing.List[str]):
+        expected_errors: typing.List[str],
+        match_all: bool = False):
 
     with pytest.raises(subprocess.CalledProcessError) as e:
         nodetool(*nodetool_args, **nodetool_kwargs)
 
     err_lines = e.value.stderr.rstrip().split('\n')
     out_lines = e.value.stdout.rstrip().split('\n')
 
-    match = False
+    match = 0
     for expected_error in expected_errors:
         if expected_error in err_lines or expected_error in out_lines:
-            match = True
+            match += 1
 
-    assert match
+    if match_all:
+        assert match == len(expected_errors)
+    else:
+        assert match > 0
```

Comment:
I know it's a corner of a test but...

Using a bool to change the meaning of a function, and with a default, leads to less-readable code. We should have check_nodetool_fails_with_all() and check_nodetool_fails_with_any(), perhaps both calling into the original as an implementation_detail.

Reply:
Ok, I will rename this to `_do_check_nodetool_fails()` and add two nicely named facades.

Reply:
Done in v2.